File: app/src/main/java/com/ott/tv/fragments/MyAccountFragment.java
package com.ott.tv.fragments;


import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ProgressBar;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.cardview.widget.CardView;
import androidx.databinding.DataBindingUtil;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;

import com.bumptech.glide.Glide;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;
import com.ott.tv.Config;
import com.ott.tv.Constants;
import com.ott.tv.databinding.FragmentMyAccountBinding;
import com.ott.tv.model.home_content.HomeContent;
import com.ott.tv.model.home_content.Video;
import com.ott.tv.model.phando.ShowWatchlist;
import com.ott.tv.model.phando.UserProfile;
import com.ott.tv.network.RetrofitClient;
import com.ott.tv.network.api.Dashboard;
import com.ott.tv.network.api.HomeApi;
import com.ott.tv.ui.activity.LoginActivity;
import com.ott.tv.utils.CMHelper;
import com.ott.tv.utils.PreferenceUtils;
import com.ott.tv.R;
import com.ott.tv.database.DatabaseHelper;
import com.ott.tv.model.ActiveStatus;
import com.ott.tv.ui.activity.LoginChooserActivity;
import com.ott.tv.utils.ToastMsg;
import com.squareup.picasso.Picasso;

import static android.content.Context.MODE_PRIVATE;

import static com.ott.tv.ui.activity.SearchActivity_Phando.progressBar;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Retrofit;

public class MyAccountFragment extends Fragment {


    private UserProfile userProfile;
    private FragmentMyAccountBinding binding;

    public MyAccountFragment() {

    }
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        binding = DataBindingUtil.inflate(
                inflater, R.layout.fragment_my_account, container, false);
        View view = binding.getRoot();

        binding.lytTopCard.setVisibility(View.GONE);
        binding.progressBar.setVisibility(View.VISIBLE);
        binding.signOutButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                signOut();
            }
        });
        getUserProfileDataFromServer();
        return view;

    }

    private void signOut() {
        if (getContext() != null && getActivity() != null) {
            DatabaseHelper databaseHelper = new DatabaseHelper(getContext());

            if (PreferenceUtils.getInstance().getAccessTokenPref(getContext()) != "") {
                SharedPreferences.Editor editor = getContext().getSharedPreferences(Constants.USER_LOGIN_STATUS, MODE_PRIVATE).edit();
                editor.putBoolean(Constants.USER_LOGIN_STATUS, false);
                editor.apply();
                databaseHelper.deleteUserData();
                PreferenceUtils.clearSubscriptionSavedData(getContext());
                PreferenceUtils.getInstance().setAccessTokenNPref(getContext(), "");
                startActivity(new Intent(getContext(), LoginChooserActivity.class));
                getActivity().finish();
            }
        }
    }


    private void getUserProfileDataFromServer() {
        binding.lytTopCard.setVisibility(View.GONE);
        binding.progressBar.setVisibility(View.VISIBLE);
        if (getActivity() != null) {
            Constants.IS_FROM_HOME = false;
            Retrofit retrofit = RetrofitClient.getRetrofitInstance();
            Dashboard api = retrofit.create(Dashboard.class);
            String accessToken = "Bearer " + PreferenceUtils.getInstance().getAccessTokenPref(getContext());
            Call<UserProfile> call = api.getUserProfileAPI(accessToken);
            call.enqueue(new Callback<UserProfile>() {
                @Override
                public void onResponse(@NotNull Call<UserProfile> call, @NotNull retrofit2.Response<UserProfile> response) {
                    if (response.isSuccessful()) {
                        binding.progressBar.setVisibility(View.GONE);
                        binding.lytTopCard.setVisibility(View.VISIBLE);
                        if (response.code() == 200 && response.body() != null) {
                            userProfile = response.body();

                            Glide.with(requireActivity())
                                    .load(userProfile.getUser().getImage())
                                    .error(R.drawable.user_image)
                                    .placeholder(R.drawable.user_image)
                                    .into(binding.imguser);

                            binding.userName.setText(userProfile.getUser().getName());
                            if (userProfile.getUser().getEmail() != null) {
                                binding.userEmailTv.setVisibility(View.VISIBLE);
                                binding.userEmailTv.setText(userProfile.getUser().getEmail());

                            }
                            if (userProfile.getIs_subscribed() ==0) {
                                binding.tvnoPlan.setVisibility(View.VISIBLE);
                                binding.userSubId.setVisibility(View.INVISIBLE);
                                binding.lytPlan.setVisibility(View.GONE);
                                binding.lytPlan2.setVisibility(View.GONE);
                                binding.myPackageName.setVisibility(View.GONE);
                                binding.myPackagePrice.setVisibility(View.GONE);
                            }
                            else
                            {
                                binding.userSubId.setVisibility(View.VISIBLE);
                                binding.myPackageName.setVisibility(View.VISIBLE);
                                binding.myPackagePrice.setVisibility(View.VISIBLE);
                                binding.userSubId.setText("Subscriber ID : " +userProfile.getCurrent_subscription().getId());
                                binding.myPackageName.setText(userProfile.getPackage_name());
                                binding.myPackagePrice.setText("Rs. " +userProfile.getCurrent_subscription().getPrice().toString());
                                binding.tvnoPlan.setVisibility(View.GONE);
                                binding.lytPlan.setVisibility(View.VISIBLE);
                                binding.lytPlan2.setVisibility(View.VISIBLE);
                                if(userProfile.getsubscription_end_date() != null){
                                    binding.myPackageDateTime.setText("Expires on " + userProfile.getsubscription_end_date());
                                }
                            }
                            if (userProfile.getUser().getMobile() != null) {
                                binding.tvMobileNumber.setText(userProfile.getUser().getMobile());
                            }
                        } else if (response.errorBody() != null) {
                            Toast.makeText(getContext(), response.errorBody().toString(), Toast.LENGTH_SHORT).show();

                        } else {
                            progressBar.setVisibility(View.GONE);
                            Toast.makeText(getContext(), "Sorry! Something went wrong. Please try again after some time", Toast.LENGTH_SHORT).show();

                            //          CMHelper.setSnackBar(requireView(), "Sorry! Something went wrong. Please try again after some time", 2);
                        }

                    } else {
                        if (response.code() == 401) {
                            //   CMHelper.setSnackBar(this.getCurrentFocus(), String.valueOf("Please Enter OTP"), 2, 10000);
                            signOut();
                            //     Toast.makeText(getContext(), "Please Login Again", Toast.LENGTH_SHORT).show();
                        } else {
                            Toast.makeText(getContext(), "Sorry! Something went wrong. Please try again after some time", Toast.LENGTH_SHORT).show();
                        }


                        Toast.makeText(getContext(), "Sorry! Something went wrong. Please try again after some time", Toast.LENGTH_SHORT).show();

                        //CMHelper.setSnackBar(requireView(), "Sorry! Something went wrong. Please try again after some time", 2);
                    }


                }
                @Override
                public void onFailure(@NonNull Call<UserProfile> call, @NonNull Throwable t) {
                    t.printStackTrace();
                    CMHelper.setSnackBar(requireView(), t.getMessage(), 2);
                    binding.progressBar.setVisibility(View.GONE);
                    binding.lytTopCard.setVisibility(View.GONE);

                }
            });
        }
    }

}

File: app/src/main/java/com/ott/tv/fragments/MyAccountFragment.kt
package com.ott.tv.fragments

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.Fragment
import com.bumptech.glide.Glide
import com.ott.tv.BuildConfig
import com.ott.tv.Constants
import com.ott.tv.R
import com.ott.tv.database.DatabaseHelper
import com.ott.tv.databinding.FragmentMyAccountBinding
import com.ott.tv.model.LanguageItem
import com.ott.tv.model.phando.UserProfile
import com.ott.tv.network.RetrofitClient
import com.ott.tv.network.api.Dashboard
import com.ott.tv.ui.activity.LanguageActivity
import com.ott.tv.ui.activity.LoginChooserActivity
import com.ott.tv.ui.activity.SearchActivity_Phando
import com.ott.tv.ui.activity.SplashScreenActivityTv
import com.ott.tv.utils.CMHelper
import com.ott.tv.utils.PreferenceUtils
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class MyAccountFragment : Fragment() {
    private var userProfile: UserProfile? = null
    private var binding: FragmentMyAccountBinding? = null
    private val TAG = SplashScreenActivityTv::class.java.simpleName
    private var Language: List<LanguageItem?>? = null

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        binding = DataBindingUtil.inflate(
            inflater, R.layout.fragment_my_account, container, false
        )
        val view = binding!!.getRoot()
        binding!!.lytTopCard.visibility = View.GONE
        binding!!.progressBar.visibility = View.VISIBLE
        binding!!.signOutButton.setOnClickListener { signOut() }
        /*binding!!.signOutButton.setOnClickListener {

            val intent = Intent(context, LanguageActivity::class.java)
            context?.startActivity(intent)

        }*/
        binding!!.languageBtn?.setOnClickListener {
            val intent = Intent(context, LanguageActivity::class.java)
            context?.startActivity(intent)
        }
        userProfileDataFromServer
        binding!!.builVersion.text = "App Version:" + BuildConfig.VERSION_CODE
        return view
    }

    private fun signOut() {
        if (context != null && activity != null) {
            val databaseHelper = DatabaseHelper(context)
            if (PreferenceUtils.getInstance().getAccessTokenPref(context) !== "") {
                val editor = context!!.getSharedPreferences(
                    Constants.USER_LOGIN_STATUS,
                    Context.MODE_PRIVATE
                ).edit()
                editor.putBoolean(Constants.USER_LOGIN_STATUS, false)
                editor.apply()
                databaseHelper.deleteUserData()
                PreferenceUtils.clearSubscriptionSavedData(context)
                PreferenceUtils.getInstance().setAccessTokenNPref(context, "")
                startActivity(Intent(context, LoginChooserActivity::class.java))
                activity!!.finish()
            }else
                {
                    val editor = context!!.getSharedPreferences(
                        Constants.USER_LOGIN_STATUS,
                        Context.MODE_PRIVATE
                    ).edit()
                    editor.putBoolean(Constants.USER_LOGIN_STATUS, false)
                    editor.apply()
                    databaseHelper.deleteUserData()
                    PreferenceUtils.clearSubscriptionSavedData(context)
                    PreferenceUtils.getInstance().setAccessTokenNPref(context, "")
                    startActivity(Intent(context, LoginChooserActivity::class.java))
                    activity!!.finish()

            }
        }
    }

    private fun fetchLanguageData(token: String) {
        if (context != null) {
            val retrofit = RetrofitClient.getRetrofitInstance()
            val api = retrofit.create(Dashboard::class.java)
            Constants.IS_FROM_HOME = false
            val accessToken = "Bearer $token"
            val call = api.getLanguage(accessToken)
            call.enqueue(object : Callback<List<LanguageItem?>> {
                override fun onResponse(
                    call: Call<List<LanguageItem?>>,
                    response: Response<List<LanguageItem?>>
                ) {
                    if (response.code() == 200) {
                        Language = response.body()
                        if (Language?.size!! > 0) {

                            // here new          setLanguageUI()
                            val selectedLanguages =
                                Language!!.filter { it!!.isLanguageSelected }.map { it!!.language }
// Join the selected languages into a comma-separated string
                            val selectedLanguagesAPI = selectedLanguages.joinToString(", ")
                            Log.i("Languages", "onResponse: --->" + selectedLanguagesAPI)
                            if (selectedLanguages.isEmpty()) {
                                binding!!.langus?.text = "No Preferences"

                            } else {
                                binding!!.langus?.text = selectedLanguagesAPI
                            }

                        } else {
                        }
                        //  homeContent.setHomeContentId(1);
                        //   homeContent.getSlider();
                        //  loadSliderRows(homeContent.getSlider().getSlideArrayList());

                        /*if (movieList.size() <= 0) {

                    }

                    for (BrowseData movie : movieList) {

                    }
                    //   movies.addAll(movieList);*/
                    } else if (response.code() == 401) {
                        signOut()
                    } else if (response.errorBody() != null) {
                        if (this != null) {
                            Toast.makeText(
                                context,
                                "sorry! Something went wrong. Please try again after some time" + response.errorBody(),
                                Toast.LENGTH_SHORT
                            ).show()
                        }
                        //  CMHelper.setSnackBar(requireView(), response.errorBody().toString(), 2);
                    } else {
                        if (context != null) {
                            Toast.makeText(
                                context,
                                "sorry! Something went wrong. Please try again after some time",
                                Toast.LENGTH_SHORT
                            ).show()
                        }
                    }
                }

                override fun onFailure(call: Call<List<LanguageItem?>>, t: Throwable) {
                    //   CMHelper.setSnackBar(requireView(), t.getMessage(), 2);
                }
            })
        }
    }

    /*
    fun callLanguage() {
        val call: Call<ArrayList<Language>> = apiService.getLanguage()
        call.enqueue(object : Callback<ArrayList<Language>> {
            override fun onResponse(
                call: Call<ArrayList<Language>>?,
                response: Response<ArrayList<Language>>?,
            ) {
                if (response == null || response.body() == null) {
                    onFailure(call, NullResponseError())
                } else {

                    val data = response.body()
                    if (data?.isNotEmpty() == true) {
                        val languageDao = AppDatabase.getInstance(getApplication())?.languageDao()
                        languageDao?.deleteAll()
                        languageDao?.insertAll(data)
                    }
                }
            }

            override fun onFailure(call: Call<ArrayList<Language>>?, t: Throwable?) {
                Log.e("", "Language not found")
            }
        })
    }
*/
    private val userProfileDataFromServer: Unit
        private get() {
            binding!!.lytTopCard.visibility = View.GONE
            binding!!.progressBar.visibility = View.VISIBLE
            if (activity != null) {
                Constants.IS_FROM_HOME = false
                val retrofit = RetrofitClient.getRetrofitInstance()
                val api = retrofit.create(Dashboard::class.java)
                val accessToken = "Bearer " + PreferenceUtils.getInstance().getAccessTokenPref(
                    context
                )
                val call = api.getUserProfileAPI(accessToken)
                call.enqueue(object : Callback<UserProfile?> {
                    override fun onResponse(
                        call: Call<UserProfile?>,
                        response: Response<UserProfile?>
                    ) {
                        if (response.isSuccessful) {
                            binding!!.progressBar.visibility = View.GONE
                            binding!!.lytTopCard.visibility = View.VISIBLE
                            if (response.code() == 200 && response.body() != null) {
                                userProfile = response.body()
                                if (activity != null) {
                                    Glide.with(activity!!)
                                        .load(userProfile!!.user.image)
                                        .error(R.drawable.user_icon_demo)
                                        .placeholder(R.drawable.user_icon_demo)
                                        .into(binding!!.imguser)
                                }
                                binding!!.userName.text = userProfile!!.user.name
                                if (userProfile!!.user.email != null) {
                                    binding!!.userEmailTv.visibility = View.VISIBLE
                                    binding!!.userEmailTv.text = userProfile!!.user.email
                                }
                                if (userProfile!!.is_subscribed == 0) {
                                    binding!!.tvnoPlan.visibility = View.VISIBLE
                                    binding!!.userSubId.visibility = View.VISIBLE
                                    binding!!.lytPlan.visibility = View.GONE
                                    binding!!.lytPlan2.visibility = View.GONE
                                    binding!!.expireDateTv.visibility = View.GONE
                                    binding!!.llMediaPlan.visibility = View.GONE
                                    binding!!.myPackageName.visibility = View.GONE
                                    binding!!.myPackagePrice.visibility = View.GONE
                                } else {
                                    binding!!.userSubId.visibility = View.VISIBLE
                                    binding!!.myPackageName.visibility = View.VISIBLE
                                    binding!!.myPackagePrice.visibility = View.VISIBLE
                                    binding!!.userSubId.text =
                                        "Subscriber ID : " + userProfile!!.current_subscription.id
                                    binding!!.myPackageName.text = userProfile!!.package_name
                                    if(userProfile!!.current_subscription!!.plan!=null){

                                        binding!!.myPackagePrice.text =userProfile!!.current_subscription!!.plan!!.currency+"  "+
                                                /*"Rs. "*/  userProfile!!.current_subscription.price.toString()


                                    }else{
                                        binding!!.myPackagePrice.text ="Subscription Plan"+"  "+
                                                /*"Rs. "*/  userProfile!!.current_subscription.price.toString()
                                    }


                                    binding!!.tvnoPlan.visibility = View.GONE
                                    binding!!.expireDateTv.visibility = View.VISIBLE
                                    binding!!.llMediaPlan.visibility = View.VISIBLE
                                    binding!!.lytPlan.visibility = View.VISIBLE
                                    binding!!.lytPlan2.visibility = View.VISIBLE
                                    if (userProfile!!.getsubscription_end_date() != null) {
                                        binding!!.myPackageDateTime.text =
                                            "Expires on " + userProfile!!.getsubscription_end_date()
                                    }
                                }
                                if (userProfile!!.user.mobile != null) {
                                    if (!userProfile!!.user.mobile.isEmpty()) {
                                        binding!!.tvMobileNumber.text = userProfile!!.user.mobile

                                    } else {
                                        binding!!.tvMobileNumber.visibility = View.GONE
                                        binding!!.viewlineSecond?.visibility = View.GONE

                                    }
                                }
                            } else if (response.errorBody() != null) {
                                Toast.makeText(
                                    context,
                                    response.errorBody().toString(),
                                    Toast.LENGTH_SHORT
                                ).show()
                            } else {
                                SearchActivity_Phando.progressBar.visibility = View.GONE
                                Toast.makeText(
                                    context,
                                    "Sorry! Something went wrong. Please try again after some time",
                                    Toast.LENGTH_SHORT
                                ).show()
                            }
                            if (userProfile!!.is_review == 1) {
                                binding!!.userSubId.visibility = View.INVISIBLE
                            } else if (userProfile!!.is_review == 0) {
                                binding!!.userSubId.visibility = View.VISIBLE
                            }
                        } else {
                            if (response.code() == 401) {
                                //   CMHelper.setSnackBar(this.getCurrentFocus(), String.valueOf("Please Enter OTP"), 2, 10000);
                                signOut()
                                //     Toast.makeText(getContext(), "Please Login Again", Toast.LENGTH_SHORT).show();
                            } else {
                                Toast.makeText(
                                    context,
                                    "Sorry! Something went wrong. Please try again after some time",
                                    Toast.LENGTH_SHORT
                                ).show()
                            }


                            //  Toast.makeText(getContext(), "Sorry! Something went wrong. Please try again after some time", Toast.LENGTH_SHORT).show();

                            //CMHelper.setSnackBar(requireView(), "Sorry! Something went wrong. Please try again after some time", 2);
                        }
                    }

                    override fun onFailure(call: Call<UserProfile?>, t: Throwable) {
                        t.printStackTrace()
                        CMHelper.setSnackBar(requireView(), t.message, 2)
                        binding!!.progressBar.visibility = View.GONE
                        binding!!.lytTopCard.visibility = View.GONE
                    }
                })
            }
        }

    override fun onResume() {
        super.onResume()
    //    fetchLanguageData(PreferenceUtils.getInstance().getAccessTokenPref(context))
    }
}